### PR TITLE
storageccl: Stop using RocksDB in GetAllRevisions, VerifyBatchRepr

### PR DESCRIPTION
**Pull request by itsbilal**
GetAllRevisions currently uses RocksDBSstFileReader.
In addition to requiring the use of RocksDB, it
is pretty heavy to use as it spins up an in-memory RocksDB
instance, ingests the SST into it, and then spins up an iterator.
This overhead can be avoided by using storage.SSTIterator,
which is based on Pebble's SST readers.

VerifyBatchRepr currently calls into Libroach to spin up
a batch iterator to compute stats on. This change does the
same dance but completely in Go-land, using a Pebble SST
iterator.

Release justification: Will land in 21.1. Cleanup change

### CODE REVIEW

**Review by dt — APPROVED** [review]
Nice!

I feel a bit bad you already went to all this effort since I'm pretty sure `VerifyBatchRepr` is only called in `evalWriteBatch` and I'm 99% sure that is dead code.

File: pkg/ccl/storageccl/revision_reader.go [hunk 1]
```diff
@@ -47,20 +47,24 @@ func GetAllRevisions(
...
 		}
-		if err := sst.Iterate(startKey, endKey, func(kv storage.MVCCKeyValue) (bool, error) {
-			if len(res) == 0 || !res[len(res)-1].Key.Equal(kv.Key.Key) {
-				res = append(res, VersionedValues{Key: kv.Key.Key})
+		defer iter.Close()
+		iter.SeekGE(storage.MVCCKey{Key: startKey})
+
+		for ; ; iter.Next() {
+			if valid, err := iter.Valid(); !valid || err != nil {
+				break
```
Comment (dt) [inline pkg/ccl/storageccl/revision_reader.go:59]: should we be returning `err` ?

**Review by petermattis — APPROVED** [review]
:lgtm:

> Reviewed 1 of 1 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/53978#-:-MGQOdIK1a7Tfsoq6Ipt:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @itsbilal and @petermattis)

___
*[pkg/ccl/storageccl/engineccl/batch_repr.go, line 25 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/53978#-MGQO1yTCT8KnfU4SRR2:-MGQO1yTCT8KnfU4SRR3:b-56nky8) ([raw file](https://github.com/cockroachdb/cockroach/blob/4c98c9e8080abff4cc48a7d12fbd6deb64200011/pkg/ccl/storageccl/engineccl/batch_repr.go#L25)):*
> ```Go
> ) (enginepb.MVCCStats, error) {
> 	// We store a 4 byte checksum of each key/value entry in the value. Make
> 	// sure the all ones in this BatchRepr validate.
> ```

`s/the all ones/all the ones/g`

**Review by itsbilal — COMMENTED** [review]
TFTRs! 

Build failure was due to reuse of unsafe keys. Fixed.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/53978#-:-MGy1-1wCxfv_2xSEeuD:b-de21xi)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @dt, @itsbilal, and @petermattis)

___
*[pkg/ccl/storageccl/revision_reader.go, line 59 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/53978#-MGQ-8yVSc3_X5D7v1fw-r2-59:-MGy1ZtB6bDbb_x2rVLB:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/4c98c9e8080abff4cc48a7d12fbd6deb64200011/pkg/ccl/storageccl/revision_reader.go#L59)):*
<details><summary><i>Previously, dt (David Taylor) wrote…</i></summary><blockquote>

should we be returning `err` ?
</blockquote></details>

Done.
___
*[pkg/ccl/storageccl/engineccl/batch_repr.go, line 25 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/53978#-MGQO1yTCT8KnfU4SRR2:-MGy1dmc2x1CGaj06VXy:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/4c98c9e8080abff4cc48a7d12fbd6deb64200011/pkg/ccl/storageccl/engineccl/batch_repr.go#L25)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary><blockquote>

`s/the all ones/all the ones/g`
</blockquote></details>

Done.

**Review by petermattis — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/53978#-:-MGyByUY-l3sYCUlwBvK:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @dt, @itsbilal, and @petermattis)

___
*[pkg/ccl/storageccl/engineccl/batch_repr.go, line 29 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/53978#-MGyNC-X5vWpt-eO4f-S:-MGyNC-X5vWpt-eO4f-T:bbtgbi5) ([raw file](https://github.com/cockroachdb/cockroach/blob/29cb28d4f6468d33f23818b57a42823045e21a97/pkg/ccl/storageccl/engineccl/batch_repr.go#L29)):*
> ```Go
> 	var kvs []storage.MVCCKeyValue
> 
> 	r, err := storage.NewRocksDBBatchReader(repr)
> ```

We should rename this `storage.NewBatchReader` at some point. And the corresponding struct to `BatchReader`.

**Comment by itsbilal** [comment]
TFTR!

bors r=petermattis

**Pull request merged**